noir-projects/noir-contracts/contracts/claim_contract/src/main.nr

```diff
@@ -38,7 +39,8 @@ contract Claim {
         // Note: Only the owner of the npk_m will be able to produce the nsk_app and compute this nullifier.
         // The nullifier is unique to the note and THIS contract because the protocol siloes all nullifiers with
         // the address of a contract it was emitted from.
-        let (_, nullifier) = proof_note.compute_note_hash_and_nullifier(&mut context);
+        let note_hash_for_consumption = compute_note_hash_for_consumption(proof_note);
+        let nullifier = proof_note.compute_nullifier(&mut context, note_hash_for_consumption);
```

Comment:
The API became worse here as we have to pass in the note hash to the compute_nullifier func even though the func itself could derive it on its own. But for now just linked an issue here as it makes sense to possibly fix once cleaning up the note interface.

Reply:
True, if we really wanted we could provide a more full version of `compute_nullifier`. But note that in this case we've called `Header::prove_note_inclusion`, which computes the note hash for consumption. So we're actually computing this twice, and if the header fn returned that value we'd be able to reuse it here.

Reply:
Returning the value from prove_note_inclusion seems like the correct approach. Expanded the comment with this info in https://github.com/AztecProtocol/aztec-packages/pull/7891/commits/de124e1f9e8f9a3408f1249603ff3eaeb04db724